src/Dropdown.php

```diff
@@ -4,359 +4,696 @@
     {
-        if ($this->tagName) {
-            return Html::closeTag($this->tagName);
-        }
+        $new = clone $this;
+        $new->cssClass = $value;
 
-        $layout = $this->prepareDropdownLayout();
+        return $new;
+    }
 
-        if (is_array($this->items)) {
-            /** @var Ul $layout */
-            return $this->renderItems($layout, $this->items);
-        }
-        /** @var CustomTag $layout */
-        return $layout
-            ->content((string) $this->items)
-            ->render();
+    /**
+     * Whether to render the dropdown in a container `<div>` tag.
+     *
+     * @param bool $value Whether to render the dropdown in a container `<div>` tag. For default, it will be rendered in
+     * a container `<div>` tag. If set to `false`, the container will not be rendered.
+     *
+     * @return self A new instance with the specified container setting.
+     */
+    public function container(bool $value): self
+    {
+        $new = clone $this;
+        $new->container = $value;
+
+        return $new;
     }
 
     /**
-     * List of menu items in the dropdown. Each array element can be either an HTML string, or an array representing a
-     * single menu with the following structure:
-     *
-     * - label: string, required, the label of the item link.
-     * - encode: bool, optional, whether to HTML-encode item label.
-     * - url: string|array, optional, the URL of the item link. This will be processed by {@see currentPath}.
-     *   If not set, the item will be treated as a menu header when the item has no sub-menu.
-     * - visible: bool, optional, whether this menu item is visible. Defaults to true.
-     * - linkOptions: array, optional, the HTML attributes of the item link.
-     * - options: array, optional, the HTML attributes of the item.
-     * - items: array, optional, the submenu items. The structure is the same as this property.
-     *   Note that Bootstrap doesn't support dropdown submenu. You have to add your own CSS styles to support it.
-     * - submenuOptions: array, optional, the HTML attributes for sub-menu container tag. If specified it will be
-     *   merged with {@see submenuOptions}.
-     *
-     * To insert divider use `-`.
+     * Sets the CSS class for the dropdown container.
+     *
+     * @param BackedEnum|string $value The CSS class for the dropdown container.
+     *
+     * @return self A new instance with the specified CSS class for the dropdown container.
      */
-    public function items(array|string|Stringable|null $value): self
+    public function containerClass(BackedEnum|string $value): self
+    {
+        $new = clone $this;
+        $new->containerClass = $value;
+
+        return $new;
+    }
+
+    /**
+     * Set the direction of the dropdown component. The following options are allowed:
+     *
+     * - `DropdownDirection::CENTERED`: Make the dropdown menu centered below the toggle with `.dropdown-center` on the
```

Comment:
Options description could be in enum only.

Reply:
done.